Kelompok 14 STARTERPACK Tugas 6 ( 6 September 2025)

Achmad Baharuddin Al Anshory

Aditya Winarto

Andi Sri Rezky Dian Batari

Elvira Dwi Anjani

Link Dataset : [link dataset](https://docs.google.com/spreadsheets/d/1cD9GivqakBxwsQKtR5pz5kOGYQWoddgr/edit?usp=sharing&ouid=106957609319676578855&rtpof=true&sd=true)

Link tersebut adalah link dataset yang digunakan untuk analisis statistik deskriptif, analisis visualisasi, serta dokumentasi insight dan hipotesis. dataset tersebut adalah dataset yang sudah bersih

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import shapiro, levene, ttest_ind, pearsonr,f_oneway
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

In [ ]:
sheet_id = '1cD9GivqakBxwsQKtR5pz5kOGYQWoddgr'
xlsx_url = f'https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=xlsx'
data = pd.read_excel(xlsx_url)
pd.DataFrame(data)

Order ID Restaurant Name         Location          Order Time  \
0     ORD001        Domino's     New York, NY 2024-01-05 18:30:00   
1     ORD002     Papa John's  Los Angeles, CA 2024-02-14 20:00:00   
2     ORD003  Little Caesars      Chicago, IL 2024-03-21 12:15:00   
3     ORD004       Pizza Hut        Miami, FL 2024-04-10 19:45:00   
4     ORD005   Marco's Pizza       Dallas, TX 2024-05-05 13:00:00   
..       ...             ...              ...                 ...   
695   ORD697       Pizza Hut      Atlanta, GA 2025-09-10 19:00:00   
696   ORD698     Papa John's        Omaha, NE 2025-09-11 20:15:00   
697   ORD699  Little Caesars   Louisville, KY 2025-09-12 19:30:00   
698   ORD700   Marco's Pizza    Milwaukee, WI 2025-09-13 18:45:00   
699   ORD701        Domino's  Albuquerque, NM 2025-09-14 20:00:00   

          Delivery Time  Delivery Duration (min) Pizza Size    Pizza Type  \
0   2024-01-05 18:45:00                       20     Medium           Veg   
1   2024-02-14 20:25:00                       25      Large       Non-Veg   
2   2024-03-21 12:35:00                       20      Small         Vegan   
3   2024-04-10 20:10:00                       25         XL  Cheese Burst   
4   2024-05-05 13:20:00                       20     Medium       Non-Veg   
..                  ...                      ...        ...           ...   
695 2025-09-10 19:30:00                       30     Medium           Veg   
696 2025-09-11 20:45:00                       30     Medium   Gluten-Free   
697 2025-09-12 20:00:00                       30         XL  Cheese Burst   
698 2025-09-13 19:15:00                       30     Medium  Thai Chicken   
699 2025-09-14 20:30:00                       30      Large      Sicilian   

     Toppings Count  Distance (km)  ... Topping Density Order Month  \
0                 3            2.5  ...        1.111111     January   
1                 4            5.0  ...        0.800000    February   
2                 2            3.0  ...        0.666667       March   
3                 5            4.5  ...        1.111111       April   
4                 3            2.0  ...        1.111111         May   
..              ...            ...  ...             ...         ...   
695               2            4.0  ...        0.500000   September   
696               3            5.0  ...        0.600000   September   
697               5            6.0  ...        0.833333   September   
698               3            4.5  ...        0.666667   September   
699               4            5.5  ...        0.727273   September   

     Payment Category  Estimated Duration (min)  Delay (min)  Is Delayed  \
0              Online                       6.0         11.6       False   
1              Online                      12.0         13.0       False   
2              Online                       7.2         12.8       False   
3             Offline                      10.8         14.2       False   
4              Online                       4.8         15.2       False   
..                ...                       ...          ...         ...   
695            Online                       9.6         20.4       False   
696            Online                      12.0         18.0       False   
697            Online                      14.4         15.6       False   
698           Offline                      10.8         19.2       False   
699            Online                      13.2         16.8       False   

    Pizza Complexity Traffic Impact  Order Hour  Restaurant Avg Time  
0                  6              2          18            30.259434  
1                 12              3          20            28.186275  
2                  2              1          13            28.844221  
3                 20              2          19            29.948454  
4                  6              3          13            30.286458  
..               ...            ...         ...           

In [ ]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 25 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   Order ID                      700 non-null    object        
 1   Restaurant Name               700 non-null    object        
 2   Location                      700 non-null    object        
 3   Order Time                    700 non-null    datetime64[ns]
 4   Delivery Time                 700 non-null    datetime64[ns]
 5   Delivery Duration (min)       700 non-null    int64         
 6   Pizza Size                    700 non-null    object        
 7   Pizza Type                    700 non-null    object        
 8   Toppings Count                700 non-null    int64         
 9   Distance (km)                 700 non-null    float64       
 10  Traffic Level                 700 non-null    object        
 11  Payment Method                70


# Statistik Deskriptif untuk setiap variable dalam hipotesis



In [ ]:
def analyze_numeric_column_direct(data, column_name):
    numerical_data = data[column_name]
    stats = numerical_data.describe()
    stats['median'] = numerical_data.median()
    stats['range'] = numerical_data.max() - numerical_data.min()
    stats['iqr'] = stats['75%'] - stats['25%']
    ordered_cols = ['count', 'mean', 'median', 'std', 'min', 'max', 'range', '25%', '50%', '75%', 'iqr']
    stats_output = stats.to_frame(name='Nilai Statistik').T
    final_stats = stats_output[[col for col in ordered_cols if col in stats_output.columns]]
    print(final_stats.to_markdown(index=False, numalign="left", stralign="left"))

In [ ]:
analyze_numeric_column_direct(data,"Delivery Duration (min)")

| count   | mean    | median   | std    | min   | max   | range   | 25%   | 50%   | 75%   | iqr   |
|:--------|:--------|:---------|:-------|:------|:------|:--------|:------|:------|:------|:------|
| 700     | 29.2786 | 25       | 9.2695 | 20    | 50    | 30      | 20    | 25    | 35    | 15    |


In [ ]:
analyze_numeric_column_direct(data,"Distance (km)")

| count   | mean    | median   | std    | min   | max   | range   | 25%   | 50%   | 75%   | iqr   |
|:--------|:--------|:---------|:-------|:------|:------|:--------|:------|:------|:------|:------|
| 700     | 4.91557 | 4.1      | 2.2907 | 2     | 10    | 8       | 3     | 4.1   | 6     | 3     |


In [ ]:
analyze_numeric_column_direct(data,"Toppings Count")

| count   | mean    | median   | std     | min   | max   | range   | 25%   | 50%   | 75%   | iqr   |
|:--------|:--------|:---------|:--------|:------|:------|:--------|:------|:------|:------|:------|
| 700     | 3.33571 | 3        | 1.18292 | 1     | 5     | 4       | 2     | 3     | 4     | 2     |


In [ ]:
def analyze_categorical_column(data, column_name):
    counts = data[column_name].value_counts()
    percentages = data[column_name].value_counts(normalize=True) * 100
    distribution_df = pd.DataFrame({
        'Frequency (Count)': counts,
        'Percentage (%)': percentages.round(2)
    })
    total_count = data[column_name].count()
    unique_count = data[column_name].nunique()
    mode_val = data[column_name].mode().iloc[0]
    print(f"=== Statistik Deskriptif untuk Kolom Kategorikal: {column_name} ===")
    print(f"Total Sampel (n): {total_count}")
    print(f"Jumlah Nilai Unik: {unique_count}")
    print(distribution_df.to_markdown(numalign="left", stralign="left"))

In [ ]:
analyze_categorical_column(data, "Is Peak Hour")

=== Statistik Deskriptif untuk Kolom Kategorikal: Is Peak Hour ===
Total Sampel (n): 700
Jumlah Nilai Unik: 2
| Is Peak Hour   | Frequency (Count)   | Percentage (%)   |
|:---------------|:--------------------|:-----------------|
| True           | 645                 | 92.14            |
| False          | 55                  | 7.86             |


In [ ]:
analyze_categorical_column(data, "Payment Category")

=== Statistik Deskriptif untuk Kolom Kategorikal: Payment Category ===
Total Sampel (n): 700
Jumlah Nilai Unik: 2
| Payment Category   | Frequency (Count)   | Percentage (%)   |
|:-------------------|:--------------------|:-----------------|
| Online             | 513                 | 73.29            |
| Offline            | 187                 | 26.71            |


In [ ]:
analyze_categorical_column(data, "Traffic Level")

=== Statistik Deskriptif untuk Kolom Kategorikal: Traffic Level ===
Total Sampel (n): 700
Jumlah Nilai Unik: 3
| Traffic Level   | Frequency (Count)   | Percentage (%)   |
|:----------------|:--------------------|:-----------------|
| High            | 263                 | 37.57            |
| Low             | 223                 | 31.86            |
| Medium          | 214                 | 30.57            |


# Uji Hipotesis



*   Hipotesis Pertama\
H0 : Rata-rata waktu pengiriman pada jam sibuk tidak lebih lama dari jam non-sibuk\
H1 : Rata-rata waktu pengiriman pada jam sibuk lebih lama dari jam non-sibuk



In [ ]:
peak = data[data['Is Peak Hour'] == True]['Delivery Duration (min)']
non_peak = data[data['Is Peak Hour'] == False]['Delivery Duration (min)']

t_stat, p_value = ttest_ind(peak, non_peak)

print("=== HASIL UJI T (Independent Samples T-Test) ===")
print(f"Nilai t-statistik: {t_stat:.3f}")
print(f"Nilai p-value: {p_value:.5f}")

if p_value < 0.05:
    print("Kesimpulan: Terdapat perbedaan signifikan antara durasi pengiriman saat jam sibuk dan jam normal.")
else:
    print("Kesimpulan: Tidak terdapat perbedaan signifikan antara durasi pengiriman saat jam sibuk dan jam normal.")

=== HASIL UJI T (Independent Samples T-Test) ===
Nilai t-statistik: 6.228
Nilai p-value: 0.00000
Kesimpulan: Terdapat perbedaan signifikan antara durasi pengiriman saat jam sibuk dan jam normal.




*   Interpretasi Hasil Uji T

Berdasarkan hasil uji t diperoleh t-statistik = 4.281 dan p-value = 0.00003, yang berarti terdapat perbedaan signifikan antara rata-rata durasi pengiriman pada jam sibuk dan jam normal. Nilai p-value yang jauh di bawah 0.05 menunjukkan bahwa perbedaan ini tidak terjadi secara kebetulan.

Hal ini mengindikasikan bahwa waktu pengiriman cenderung lebih lama saat jam sibuk dibandingkan dengan jam normal. Faktor kemacetan dan tingginya volume pesanan kemungkinan besar memengaruhi efisiensi pengiriman. Hasil ini dapat menjadi dasar evaluasi bagi pihak restoran untuk menyesuaikan estimasi waktu atau menambah tenaga kurir pada jam sibuk.

Hipotesis Kedua\
H0 : Rata-rata waktu pengiriman sama untuk semua level traffic\
H1 : Setidaknya ada satu level traffic yang berbeda rata-rata waktu pengirimannya


In [ ]:
# Define kolom
DURASI_KOLOM = 'Delivery Duration (min)'
TRAFFIC_KOLOM = 'Traffic Level'

# Extract groups berdasarkan traffic level
low_traffic = data[data[TRAFFIC_KOLOM] == 'Low'][DURASI_KOLOM]
medium_traffic = data[data[TRAFFIC_KOLOM] == 'Medium'][DURASI_KOLOM]
high_traffic = data[data[TRAFFIC_KOLOM] == 'High'][DURASI_KOLOM]

# Uji ANOVA
f_statistic, p_value = f_oneway(low_traffic, medium_traffic, high_traffic)
print("-" * 50)
print(f"F-Statistic: {f_statistic:.4f}")
print(f"P-Value: {p_value:.5f}")
print("-" * 50)

# Uji hipotesis
alpha = 0.05
if p_value < alpha:
    print("Kesimpulan: Tolak H0")
    print("Setidaknya ada satu level traffic yang berbeda rata-rata waktu pengirimannya.")
else:
    print("Kesimpulan: Gagal Tolak H0")
    print("Rata-rata waktu pengiriman sama untuk semua level traffic.")

--------------------------------------------------
F-Statistic: 252.6689
P-Value: 0.00000
--------------------------------------------------
Kesimpulan: Tolak H0
Setidaknya ada satu level traffic yang berbeda rata-rata waktu pengirimannya.




*   Interpretasi Hasil Uji ANOVA

Berdasarkan hasil uji ANOVA satu arah yang terlihat pada gambar, diperoleh F-Statistic = 252.6689 dan p-value = 0.00000, yang berarti terdapat perbedaan signifikan pada rata-rata durasi pengiriman antar berbagai level traffic (Low, Medium, High). Nilai p-value yang jauh di bawah 0.05 menunjukkan bahwa perbedaan ini tidak terjadi secara kebetulan, melainkan secara statistik sangat signifikan dengan tingkat kepercayaan yang tinggi.

Hal ini mengindikasikan bahwa level traffic memiliki pengaruh yang sangat kuat terhadap waktu pengiriman pizza. Kondisi lalu lintas yang berbeda-beda (rendah, sedang, tinggi) secara konsisten mempengaruhi efisiensi pengiriman, kemungkinan karena faktor kemacetan, kecepatan pergeraan kendaraan, dan aksesibilitas rute yang bervariasi. Hasil ini memberikan dasar yang kuat bagi pihak restoran untuk mengimplementasikan strategi pengiriman yang adaptif berdasarkan kondisi traffic real-time, seperti penyesuaian estimasi waktu pengiriman, optimasi rute dinamis, dan alokasi sumber daya kurir yang disesuaikan dengan level traffic untuk menjaga konsistensi layanan dan kepuasan pelanggan di berbagai kondisi lalu lintas.





*   Hipotesis Ketiga\
H0 : Tidak ada hubungan linear antara jumlah topping dengan waktu pengiriman\
H1 : Ada hubungan linear positif antara jumlah topping dengan waktu pengiriman




In [ ]:
x = data['Toppings Count']
y = data['Delivery Duration (min)']

r_value, p_value = pearsonr(x, y)

print("=== HASIL UJI KORELASI PEARSON ===")
print(f"Koefisien Korelasi (r): {r_value:.3f}")
print(f"Nilai p-value: {p_value:.5f}")

if p_value < 0.05:
    print("Kesimpulan: Terdapat hubungan linier positif antara jumlah toping dan waktu pengiriman.")
else:
    print("Kesimpulan: Tidak terdapat hubungan linier positif antara jumlah toping dan waktu pengiriman.")


=== HASIL UJI KORELASI PEARSON ===
Koefisien Korelasi (r): 0.558
Nilai p-value: 0.00000
Kesimpulan: Terdapat hubungan linier positif antara jumlah toping dan waktu pengiriman.




*   Interpretasi Uji Korelasi

Berdasarkan hasil uji korelasi Pearson, diperoleh koefisien korelasi (r) sebesar 0.558 dengan p-value = 0.00000, yang menunjukkan adanya hubungan positif yang kuat dan signifikan antara jumlah topping (Toppings Count) dengan durasi pengiriman (Delivery Duration (min)). Artinya, semakin banyak topping pada pizza yang dipesan, maka waktu pengiriman pizza juga akan semakin lama. Nilai p-value yang lebih kecil dari 0.05 menegaskan bahwa hubungan tersebut tidak terjadi secara kebetulan, melainkan secara statistik signifikan.

Hasil ini memberikan gambaran bahwa kompleksitas topping menjadi salah satu faktor penting dalam proses pengiriman pizza. Dengan mengetahui kekuatan hubungan ini, pihak restoran dapat memperkirakan waktu pengiriman yang lebih akurat berdasarkan kompleksitas pesanan serta mengoptimalkan proses persiapan di dapur agar durasi pengiriman dapat diminimalkan. Pizza dengan topping yang lebih banyak memerlukan waktu persiapan yang lebih lama, yang pada akhirnya berdampak pada total waktu pengiriman. Temuan ini dapat digunakan sebagai dasar dalam meningkatkan perencanaan produksi, manajemen inventori topping, dan estimasi waktu pengiriman yang lebih realistis untuk meningkatkan kepuasan pelanggan.




*   Hipotesis Keempat\
H0 : Rata-rata waktu pengiriman dengan pembayaran digital tidak lebih cepat dari pembayaran tunai\
H1 : Rata-rata waktu pengiriman dengan pembayaran digital lebih cepat dari pembayaran tunai




In [ ]:
online = data[data['Payment Category'] == 'Online']['Delivery Duration (min)']
offline = data[data['Payment Category'] == 'Offline']['Delivery Duration (min)']
t_stat, p_value = ttest_ind(online, offline)

print("\n=== HASIL UJI T (Independent Samples T-Test) ===")
print(f"Nilai t-statistik: {t_stat:.3f}")
print(f"Nilai p-value: {p_value:.5f}")

# Statistical conclusion
alpha = 0.05
if p_value < alpha:
    if online.mean() > offline.mean():
        conclusion = "Terdapat perbedaan signifikan: pembayaran online memiliki waktu pengiriman yang lebih lama dibandingkan pembayaran offline."
    else:
        conclusion = "Terdapat perbedaan signifikan: pembayaran offline memiliki waktu pengiriman yang lebih lama dibandingkan pembayaran online."
    print(f"Kesimpulan: {conclusion}")
else:
    print("Kesimpulan: Tidak terdapat perbedaan signifikan pada waktu pengiriman antara pembayaran online dan offline.")



=== HASIL UJI T (Independent Samples T-Test) ===
Nilai t-statistik: -0.460
Nilai p-value: 0.64591
Kesimpulan: Tidak terdapat perbedaan signifikan pada waktu pengiriman antara pembayaran online dan offline.




*   interpretasi Hasil Uji T

Berdasarkan hasil uji t (Independent Samples T-Test), diperoleh t-statistik = -0.460 dan p-value = 0.64591, yang berarti tidak terdapat perbedaan signifikan antara rata-rata durasi pengiriman pada pembayaran online dan offline. Nilai p-value yang jauh di atas 0.05 menunjukkan bahwa perbedaan yang ada kemungkinan besar terjadi secara kebetulan dan tidak memiliki signifikansi statistik.

Hal ini mengindikasikan bahwa metode pembayaran (online maupun offline) tidak mempengaruhi waktu pengiriman pizza secara signifikan. Sistem operasional dan logistik pengiriman tampaknya sudah berjalan dengan konsisten terlepas dari jenis pembayaran yang dipilih pelanggan. Hasil ini menunjukkan bahwa pihak restoran telah berhasil mengelola proses pengiriman secara efisien untuk kedua kategori pembayaran. Temuan ini dapat memberikan kepercayaan kepada pelanggan bahwa tidak ada bias waktu pengiriman berdasarkan metode pembayaran yang mereka pilih, sehingga pelanggan dapat memilih metode pembayaran sesuai preferensi tanpa khawatir akan mempengaruhi kecepatan pengiriman.







*   Hipotesis Kelima\
 H0 : Tidak ada hubungan linear antara jarak dengan waktu pengiriman\
 H1 : Ada hubungan linear positif antara jarak dengan waktu pengiriman




In [ ]:
x = data['Distance (km)']
y = data['Delivery Duration (min)']

r_value, p_value = pearsonr(x, y)

print("=== HASIL UJI KORELASI PEARSON ===")
print(f"Koefisien Korelasi (r): {r_value:.3f}")
print(f"Nilai p-value: {p_value:.5f}")

if p_value < 0.05:
    print("Kesimpulan: Terdapat hubungan signifikan antara jarak dan durasi pengiriman.")
else:
    print("Kesimpulan: Tidak terdapat hubungan signifikan antara jarak dan durasi pengiriman.")



*   interpretasi Hasil Korelasi



Berdasarkan hasil uji korelasi Pearson diperoleh koefisien korelasi (r) sebesar 0.932 dengan p-value = 0.00000, yang menunjukkan adanya hubungan positif yang sangat kuat dan signifikan antara jarak pengiriman (Distance (km)) dengan durasi pengiriman (Delivery Duration (min)). Artinya, semakin jauh jarak yang ditempuh oleh kurir, maka waktu pengiriman pizza juga akan semakin lama. Nilai p-value yang lebih kecil dari 0.05 menegaskan bahwa hubungan tersebut tidak terjadi secara kebetulan, melainkan secara statistik signifikan.

Hasil ini memberikan gambaran bahwa faktor jarak menjadi salah satu penentu utama dalam efisiensi pengiriman pizza. Dengan mengetahui kekuatan hubungan ini, pihak restoran dapat memperkirakan waktu pengiriman lebih akurat serta mengoptimalkan rute agar durasi pengiriman dapat diminimalkan. Temuan ini juga dapat digunakan sebagai dasar dalam meningkatkan layanan logistik dan kepuasan pelanggan.